torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3550,14 +3550,35 @@ def sample_inputs_expand_as(op_info, device, dtype, requires_grad, **kwargs):
              )
 
     def generator():
-        for case in cases:
-            shape, shape_other = case
+        for shape, shape_other in cases:
             yield(SampleInput(make_arg(shape, requires_grad=requires_grad),
                               args=(make_arg(shape_other, requires_grad=False), )))
 
     return list(generator())
 
 
+def sample_inputs_where(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, dtype=dtype, device=device, requires_grad=requires_grad)
+    make_bool_mask = partial(make_tensor, dtype=torch.bool, device=device, requires_grad=requires_grad)
+
+    WhereCases = collections.namedtuple("WhereCases",
+                                        ["shape", "mask_shape", "other_shape", "broadcasts_input"])
+
+    cases = (WhereCases(shape=(M, M), mask_shape=(M, M), other_shape=(M, M), broadcasts_input=False),
```

Comment:
Too verbose? Or easy to read?

Reply:
It is easy to read but at the point this is written why not just do:

```
yield SampleInput(make_arg(M, M), args=(make_bool_mask(M, M), make_arg(M, M)))
```

instead of each case?

Reply:
```python
cases = (((M, M), (M, M), (M, M), False),
             ((M, 1, M), (M, M), (M, M, 1), True),
             ((), (), (), False),
             ((M, 1, M), (), (M, M, 1), True),
             ((), (M, M), (), True),)

def generator():
    for shape, mask_shape, other_shape, broadcasts_input in cases:
            yield SampleInput(make_arg(shape),
                              args=(make_bool_mask(mask_shape), make_arg(other_shape)),
                              broadcasts_input=broadcasts_input)
```
Is it ok to stick to usual `cases` as tuple. Even `yield` for every sample input is very verbose and hard to extend the cases if need be (ever).